lib/tasks/reports/find_broken_related_links.rake

```diff
@@ -0,0 +1,58 @@
+namespace :scihist do
+  namespace :reports do
+    desc """
+      # bundle exec rake scihist:reports:find_broken_related_links 
+    """
+    task :find_broken_related_links => :environment do
+      # a) Report related links that are, or redirect to, a broken URL.
+
+      # b) In addition, if the final URL of the rediect chain ends in one of these endings,
+      # we want to report it as a problem:
+      bad_endings = ['stories', 'magazine', 'distillations-pod']
+
+      Kithe::Indexable.index_with(batching: true) do
+        Work.find_each do |work|
+          next if work.related_link.length == 0
+          work.related_link.each do |link|
+            urls_followed = [link.url]
+
+            # Follow the redirect chain until we reach a 200:
+            while response = HTTP.get(urls_followed.last)
+
+
+              if response.status.to_i == 301
+                # construct the next URL in the redirect chain
+                if response.headers["Location"].start_with? 'http'
+                  urls_followed.append response.headers["Location"]
+                else # relative URL
+                  parsed_url = URI(urls_followed.last)
+                  urls_followed.append(
+                    parsed_url.scheme + "://" + parsed_url.hostname  +
+                    response.headers["Location"]
+                  )
+                end
```

Comment:
You can just tell http-rb to follow redirects, instead of doing it manually. Then you just get the terminal response as output. Might make for a lot simpler code?

https://github.com/httprb/http/wiki/Redirects

    HTTP.follow.get(url)

Reply:
That's much better yeah. Done.